samples/interactive-tutorials/product/update_product.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2021 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# [START retail_update_product]
+# Update product in a catalog using Retail API
+#
+import os
+import random
+import string
+
+from google.api_core.client_options import ClientOptions
+from google.cloud.retail_v2 import PriceInfo, Product, ProductServiceClient, \
+    UpdateProductRequest
+from google.cloud.retail_v2.types import product
+
+from setup.setup_cleanup import create_product, delete_product
+
+# from google.protobuf.field_mask_pb2 import FieldMask
+# from google.protobuf.field_mask_pb2 import FieldMask
+
+
+project_number = os.getenv('GOOGLE_CLOUD_PROJECT_NUMBER')
+default_branch_name = "projects/" + project_number + "/locations/global/catalogs/default_catalog/branches/default_branch"
+endpoint = "retail.googleapis.com"
+generated_product_id = ''.join(random.sample(string.ascii_lowercase, 8))
+
+
+# get product service client
+def get_product_service_client():
+    client_options = ClientOptions(endpoint)
+    return ProductServiceClient(client_options=client_options)
+
+
+# generate product for update
+def generate_product_for_update(product_id: str) -> Product:
+    price_info = PriceInfo()
+    price_info.price = 20.0
+    price_info.original_price = 25.5
+    price_info.currency_code = "EUR"
+    return product.Product(
+        id=product_id,
+        name='projects/' + project_number + '/locations/global/catalogs/default_catalog/branches/default_branch/products/' + product_id,
+        title='Updated Nest Mini',
+        type_=product.Product.Type.PRIMARY,
```

Comment:
type_ -> type

Reply:
the request field name is type_

Reply:
the request field name is "type_" if I try to use "type" the error is thrown:

raise ValueError(
ValueError: Unknown field for Product: type